client/my-sites/domains/domain-management/dns/txt-record.jsx

```diff
@@ -63,7 +63,10 @@ class TxtRecord extends React.Component {
 						<FormInputValidation text={ translate( 'TXT Record has non-ASCII data' ) } isWarning />
 					) }
 					{ ! isDataValid && (
-						<FormInputValidation text={ translate( 'Invalid TXT Record' ) } isError />
+						<FormInputValidation
+							text={ translate( 'TXT records may not exceed 255 characters' ) }
```

Comment:
Shouldn't we have an explicit/separate check for the length here? I'd like the validating code to be somehow involved in telling us what the issue is -- given that `isValid` is a prop, it doesn't feel right to make an assumption here about what the reason is.

Reply:
I think the code feels over-engineered _and_ somehow lacking in some other regards. I created a method in the component, since I needed another message for empty values.

You are right about the assumptions, but the only other solution would be to change how the validation works and make it return the reason as well, which I think is out of scope and perhaps not worth it right now.

Reply:
💯 agreed -- this was going to be my proposal for it as well for now. Doing more than this would be overkill.